[BUG]: #867 Makes Tests Fail Under Ansible-Test

### Bug Description

I'm currently working on upgrading `ansible-collection-dellemc-openmanage` in Fedora, but I'm running into issues running the tests.

Fedora runs the tests for Ansible collections using `ansible-test units`, which after updating from 9.12.0 to 9.12.1 is failing all tests listed later in this ticket.

After looking through the changeset for 9.12.0, it seems that #867 is the suspect commit. If I revert that commit and run the tests, they all pass.

Interestingly, this seems to only be the case under `ansible-test`. If you run the tests using `tox` (like CI does) then they all still pass.

### Component or Module Name

N/A

### Ansible Version

Ansible 2.18

### Python Version

Python 3.13

### iDRAC/OME/OME-M version

N/A

### Operating System

Fedora Linux 42

### Playbook Used

N/A

### Logs

<details>

```python
============================= test session starts ==============================
platform linux -- Python 3.13.5, pytest-8.4.1, pluggy-1.6.0
rootdir: /home/daniel/Git/Fedora/ansible_collections/dellemc/openmanage
configfile: venv/lib/python3.13/site-packages/ansible_test/_data/pytest/config/default.ini
plugins: xdist-3.7.0, mock-3.14.1, cov-6.2.1, sugar-1.0.0, plus-0.8.1, ansible-25.6.3
created: 16/16 workers
16 workers [2906 items]

........................................................................ [  2%]
...........F............................................................ [  4%]
........................................................................ [  7%]
.........................................................F.............. [  9%]
........................................................................ [ 12%]
........................................................................ [ 14%]
........................................................................ [ 17%]
.................................................F...................... [ 19%]
.......................................................................F [ 22%]
........................................................................ [ 24%]
........................................................................ [ 27%]
........................................................................ [ 29%]
........................................................................ [ 32%]
........................................................................ [ 34%]
........................................................................ [ 37%]
........................................................................ [ 39%]
........................................................................ [ 42%]
........................................................................ [ 44%]
........................................................................ [ 47%]
........................................................................ [ 49%]
........................................................................ [ 52%]
........................................................................ [ 54%]
........................................................................ [ 56%]
........................................................................ [ 59%]
........................................................................ [ 61%]
........................................................................ [ 64%]
........................................................................ [ 66%]
........................................................................ [ 69%]
........................................................................ [ 71%]
........................................................................ [ 74%]
........................................................................ [ 76%]
........................................................................ [ 79%]
........................................................................ [ 81%]
........................................................................ [ 84%]
........................................................................ [ 86%]
........................................................................ [ 89%]
........................................................................ [ 91%]
........................................................................ [ 94%]
..................FF........F........................................... [ 96%]
........................................................................ [ 99%]
..........................                                               [100%]
=================================== FAILURES ===================================
___ TestLcJobStatus.test_get_lifecycle_controller_job_status_info_invalid_id ___
[gw8] linux -- Python 3.13.5 /home/daniel/Git/Fedora/ansible_collections/dellemc/openmanage/venv/bin/python

self = <ansible_collections.dellemc.openmanage.tests.unit.plugins.modules.test_idrac_lifecycle_controller_job_status_info.TestLcJobStatus object at 0x7fc6a93f3ce0>
idrac_default_args = {'_ansible_check_mode': False, '_ansible_keep_remote_files': False, '_ansible_remote_tmp': '/tmp', 'ca_path': '/path/to/ca_cert.pem', ...}
idrac_mock = <MagicMock name='mock.job_mgr' id='140491203743760'>
mocker = <pytest_mock.plugin.MockerFixture object at 0x7fc6aaff2cf0>

    def test_get_lifecycle_controller_job_status_info_invalid_id(
            self,
            idrac_default_args,
            idrac_mock,
            mocker):
        idrac_default_args.update({"job_id": "job_id"})
        mocker.patch(MODULE_PATH + "idrac_lifecycle_controller_job_status_info.IDRACFirmwareInfo.is_omsdk_required",
                     return_value=False)
        idrac_mock.invoke_request.return_value.status_code = 200
        response1 = "Job ID is invalid"
        mocker.patch(MODULE_PATH + "idrac_lifecycle_controller_job_status_info.IDRACLifecycleControllerJobStatusInfo.get_lifecycle_controller_job_status_info",
                     return_value=response1)
        result = self._run_module(idrac_default_args)
>       assert result == {
            "msg": "Successfully fetched the job info.",
            "job_info": {},
            "changed": False}
E       assert {'changed': F...IPv6 address"} == {'changed': F...he job info.'}
E         
E         Omitting 1 identical items, use -vv to show
E         Differing items:
E         {'msg': "'idrac_ip' does not appear to be an IPv4 or IPv6 address"} != {'msg': 'Successfully fetched the job info.'}
E         Left contains 1 more item:
E         {'failed': True}
E         Right contains 1 more item:...
E         
E         ...Full output truncated (2 lines hidden), use '-vv' to show

tests/unit/plugins/modules/test_idrac_lifecycle_controller_job_status_info.py:232: AssertionError
_ TestDiagnosticsType.test_idrac_diagnostics_main_exception_handling_case[URLError] _
[gw6] linux -- Python 3.13.5 /home/daniel/Git/Fedora/ansible_collections/dellemc/openmanage/venv/bin/python

self = <ansible_collections.dellemc.openmanage.tests.unit.plugins.modules.test_idrac_diagnostics.TestDiagnosticsType object at 0x7fbeb077b110>
exc_type = <class 'urllib.error.URLError'>
mocker = <pytest_mock.plugin.MockerFixture object at 0x7fbeb223ecf0>
idrac_default_args = {'_ansible_check_mode': False, '_ansible_keep_remote_files': False, '_ansible_remote_tmp': '/tmp', 'ca_path': '/path/to/ca_cert.pem', ...}

    @pytest.mark.parametrize("exc_type",
                             [URLError, HTTPError, SSLValidationError, ConnectionError, TypeError, ValueError])
    def test_idrac_diagnostics_main_exception_handling_case(self, exc_type, mocker, idrac_default_args):
        idrac_default_args.update({"run": True})
        # Scenario 1: HTTPError with message id SYS099
        json_str = to_text(json.dumps({"error": {MESSAGE_EXTENDED: [
            {
                'MessageId': "SYS099",
                "Message": "Error"
            }
        ]}}))
        if exc_type in [HTTPError, SSLValidationError]:
            mocker.patch(MODULE_PATH + RUN_EXEC_FUNC,
                         side_effect=exc_type(HTTPS_PATH, 400,
                                              HTTP_ERROR,
                                              {"accept-type": APPLICATION_JSON},
                                              StringIO(json_str)))
        else:
            mocker.patch(MODULE_PATH + RUN_EXEC_FUNC,
                         side_effect=exc_type('test'))
        result = self._run_module(idrac_default_args)
        if exc_type == URLError:
>           assert result['unreachable'] is True
                   ^^^^^^^^^^^^^^^^^^^^^
E           KeyError: 'unreachable'

tests/unit/plugins/modules/test_idrac_diagnostics.py:1008: KeyError
_ TestSupportAssistType.test_idrac_support_assist_main_exception_handling_case[URLError] _
[gw14] linux -- Python 3.13.5 /home/daniel/Git/Fedora/ansible_collections/dellemc/openmanage/venv/bin/python

self = <ansible_collections.dellemc.openmanage.tests.unit.plugins.modules.test_idrac_support_assist.TestSupportAssistType object at 0x7f1abc62f890>
exc = <class 'urllib.error.URLError'>
mocker = <pytest_mock.plugin.MockerFixture object at 0x7f1abc597a10>
idrac_default_args = {'_ansible_check_mode': False, '_ansible_keep_remote_files': False, '_ansible_remote_tmp': '/tmp', 'ca_path': '/path/to/ca_cert.pem', ...}

    @pytest.mark.parametrize("exc",
                             [URLError, HTTPError, SSLValidationError, ConnectionError, TypeError, ValueError])
    def test_idrac_support_assist_main_exception_handling_case(self, exc, mocker, idrac_default_args):
        idrac_default_args.update(
            {"run": True, "data_collector": ['gpu_logs'], "export": False})
        # Scenario 1: HTTPError with message id SRV095
        error_string = to_text(json.dumps({"error": {MESSAGE_EXTENDED: [
            {
                'MessageId': "SRV095",
                "Message": HTTP_ERROR
            }
        ]}}))
        if exc in [HTTPError, SSLValidationError]:
            mocker.patch(MODULE_PATH + RUN_EXEC_FUNC,
                         side_effect=exc(HTTPS_PATH, 400,
                                         HTTP_ERROR,
                                         {"accept-type": APPLICATION_JSON},
                                         StringIO(error_string)))
        else:
            mocker.patch(MODULE_PATH + RUN_EXEC_FUNC,
                         side_effect=exc('test'))
        res_out = self._run_module(idrac_default_args)
        if exc == URLError:
>           assert res_out['unreachable'] is True
                   ^^^^^^^^^^^^^^^^^^^^^^
E           KeyError: 'unreachable'

tests/unit/plugins/modules/test_idrac_support_assist.py:1270: KeyError
_____ TestLcJobStatus.test_get_lifecycle_controller_job_status_info_case02 _____
[gw7] linux -- Python 3.13.5 /home/daniel/Git/Fedora/ansible_collections/dellemc/openmanage/venv/bin/python

self = <ansible_collections.dellemc.openmanage.tests.unit.plugins.modules.test_idrac_lifecycle_controller_job_status_info.TestLcJobStatus object at 0x7f523a437bb0>
idrac_default_args = {'_ansible_check_mode': False, '_ansible_keep_remote_files': False, '_ansible_remote_tmp': '/tmp', 'ca_path': '/path/to/ca_cert.pem', ...}
idrac_mock = <MagicMock name='mock.job_mgr' id='139991125512208'>
mocker = <pytest_mock.plugin.MockerFixture object at 0x7f52400becf0>

    def test_get_lifecycle_controller_job_status_info_case02(
            self,
            idrac_default_args,
            idrac_mock,
            mocker):
        idrac_default_args.update({"job_id": "job_id"})
        mocker.patch(MODULE_PATH + "idrac_lifecycle_controller_job_status_info.IDRACFirmwareInfo.is_omsdk_required",
                     return_value=False)
        idrac_mock.invoke_request.return_value.status_code = 200
        response1 = MagicMock()
        response1.json_data = {"status": "Success"}
        mocker.patch(MODULE_PATH + "idrac_lifecycle_controller_job_status_info.IDRACLifecycleControllerJobStatusInfo.get_lifecycle_controller_job_status_info",
                     return_value=response1)
        mocker.patch(MODULE_PATH + "idrac_lifecycle_controller_job_status_info.IDRACLifecycleControllerJobStatusInfo.transform_job_status_data",
                     return_value={"status": "Success"})
        result = self._run_module(idrac_default_args)
>       assert result == {
            "job_info": {"status": "Success"},
            "msg": "Successfully fetched the job info.",
            "changed": False}
E       assert {'changed': F...IPv6 address"} == {'changed': F...he job info.'}
E         
E         Omitting 1 identical items, use -vv to show
E         Differing items:
E         {'msg': "'idrac_ip' does not appear to be an IPv4 or IPv6 address"} != {'msg': 'Successfully fetched the job info.'}
E         Left contains 1 more item:
E         {'failed': True}
E         Right contains 1 more item:...
E         
E         ...Full output truncated (2 lines hidden), use '-vv' to show

tests/unit/plugins/modules/test_idrac_lifecycle_controller_job_status_info.py:214: AssertionError
______________ TestLcStatus.test_main_get_lcstatus_success_case00 ______________
[gw8] linux -- Python 3.13.5 /home/daniel/Git/Fedora/ansible_collections/dellemc/openmanage/venv/bin/python

    def main():
        specs = {}
        specs.update(idrac_auth_params)
        module = AnsibleModule(
            argument_spec=specs,
            supports_check_mode=True)
        try:
>           with iDRACRedfishAPI(module.params) as idrac:
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

plugins/modules/idrac_lifecycle_controller_status_info.py:111: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
plugins/module_utils/idrac_redfish.py:119: in __init__
    gen_details = self.get_server_generation
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
plugins/module_utils/idrac_redfish.py:224: in get_server_generation
    response = self.invoke_request(MANAGER_URI, 'GET')
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
plugins/module_utils/idrac_redfish.py:187: in invoke_request
    resp = open_url(url, data=data, **url_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/tmp/ansible-test-0stq_dwi/ansible/module_utils/urls.py:995: in open_url
    return Request().open(method, url, data=data, headers=headers, use_proxy=use_proxy,
/tmp/ansible-test-0stq_dwi/ansible/module_utils/urls.py:843: in open
    url, auth_headers, auth_handlers = _configure_auth(url, url_username, url_password, use_gssapi, force_basic_auth, use_netrc)
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/tmp/ansible-test-0stq_dwi/ansible/module_utils/urls.py:640: in _configure_auth
    parsed = urlparse(url)
             ^^^^^^^^^^^^^
/usr/lib64/python3.13/urllib/parse.py:395: in urlparse
    splitresult = urlsplit(url, scheme, allow_fragments)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib64/python3.13/urllib/parse.py:516: in urlsplit
    _check_bracketed_netloc(netloc)
/usr/lib64/python3.13/urllib/parse.py:454: in _check_bracketed_netloc
    _check_bracketed_host(hostname)
/usr/lib64/python3.13/urllib/parse.py:463: in _check_bracketed_host
    ip = ipaddress.ip_address(hostname) # Throws Value Error if not IPv6 or IPv4
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

address = 'idrac_ip'

    def ip_address(address):
        """Take an IP string/int and return an object of the correct type.
    
        Args:
            address: A string or integer, the IP address.  Either IPv4 or
              IPv6 addresses may be supplied; integers less than 2**32 will
              be considered to be IPv4 by default.
    
        Returns:
            An IPv4Address or IPv6Address object.
    
        Raises:
            ValueError: if the *address* passed isn't either a v4 or a v6
              address
    
        """
        try:
            return IPv4Address(address)
        except (AddressValueError, NetmaskValueError):
            pass
    
        try:
            return IPv6Address(address)
        except (AddressValueError, NetmaskValueError):
            pass
    
>       raise ValueError(f'{address!r} does not appear to be an IPv4 or IPv6 address')
E       ValueError: 'idrac_ip' does not appear to be an IPv4 or IPv6 address

/usr/lib64/python3.13/ipaddress.py:54: ValueError

During handling of the above exception, another exception occurred:

self = <ansible_collections.dellemc.openmanage.tests.unit.plugins.modules.test_idrac_lifecycle_controller_status_info.TestLcStatus object at 0x7fc6a94b2490>
idrac_default_args = {'_ansible_check_mode': False, '_ansible_keep_remote_files': False, '_ansible_remote_tmp': '/tmp', 'ca_path': '/path/to/ca_cert.pem', ...}
mocker = <pytest_mock.plugin.MockerFixture object at 0x7fc6a84b0ec0>

    def test_main_get_lcstatus_success_case00(self,
                                              idrac_default_args,
                                              mocker):
        mocker.patch(MODULE_PATH + "idrac_lifecycle_controller_status_info.IDRACInfo.get_idrac_hw_model",
                     return_value="9")
        lcstatus = "Ready"
        mocker.patch(MODULE_PATH + "idrac_lifecycle_controller_status_info.IDRACLifecycleControllerStatusInfo.get_lifecycle_controller_status_info",
                     return_value=lcstatus)
>       result = self._run_module(idrac_default_args)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

tests/unit/plugins/modules/test_idrac_lifecycle_controller_status_info.py:60: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/unit/plugins/modules/common.py:46: in _run_module
    self.module.main()
plugins/modules/idrac_lifecycle_controller_status_info.py:128: in main
    module.fail_json(msg=str(e))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (<ansible.module_utils.basic.AnsibleModule object at 0x7fc6a84b1d30>,)
kwargs = {'failed': True, 'msg': "'idrac_ip' does not appear to be an IPv4 or IPv6 address"}

    def fail_json(*args, **kwargs):
        kwargs['failed'] = True
>       raise AnsibleFailJson(kwargs)
E       ansible_collections.dellemc.openmanage.tests.unit.plugins.modules.utils.AnsibleFailJson: {'msg': "'idrac_ip' does not appear to be an IPv4 or IPv6 address", 'failed': True}

tests/unit/plugins/modules/utils.py:38: AnsibleFailJson
______________ TestLcStatus.test_main_get_lcstatus_success_case01 ______________
[gw8] linux -- Python 3.13.5 /home/daniel/Git/Fedora/ansible_collections/dellemc/openmanage/venv/bin/python

    def main():
        specs = {}
        specs.update(idrac_auth_params)
        module = AnsibleModule(
            argument_spec=specs,
            supports_check_mode=True)
        try:
>           with iDRACRedfishAPI(module.params) as idrac:
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

plugins/modules/idrac_lifecycle_controller_status_info.py:111: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
plugins/module_utils/idrac_redfish.py:119: in __init__
    gen_details = self.get_server_generation
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
plugins/module_utils/idrac_redfish.py:224: in get_server_generation
    response = self.invoke_request(MANAGER_URI, 'GET')
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
plugins/module_utils/idrac_redfish.py:187: in invoke_request
    resp = open_url(url, data=data, **url_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/tmp/ansible-test-0stq_dwi/ansible/module_utils/urls.py:995: in open_url
    return Request().open(method, url, data=data, headers=headers, use_proxy=use_proxy,
/tmp/ansible-test-0stq_dwi/ansible/module_utils/urls.py:843: in open
    url, auth_headers, auth_handlers = _configure_auth(url, url_username, url_password, use_gssapi, force_basic_auth, use_netrc)
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/tmp/ansible-test-0stq_dwi/ansible/module_utils/urls.py:640: in _configure_auth
    parsed = urlparse(url)
             ^^^^^^^^^^^^^
/usr/lib64/python3.13/urllib/parse.py:395: in urlparse
    splitresult = urlsplit(url, scheme, allow_fragments)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib64/python3.13/urllib/parse.py:516: in urlsplit
    _check_bracketed_netloc(netloc)
/usr/lib64/python3.13/urllib/parse.py:454: in _check_bracketed_netloc
    _check_bracketed_host(hostname)
/usr/lib64/python3.13/urllib/parse.py:463: in _check_bracketed_host
    ip = ipaddress.ip_address(hostname) # Throws Value Error if not IPv6 or IPv4
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

address = 'idrac_ip'

    def ip_address(address):
        """Take an IP string/int and return an object of the correct type.
    
        Args:
            address: A string or integer, the IP address.  Either IPv4 or
              IPv6 addresses may be supplied; integers less than 2**32 will
              be considered to be IPv4 by default.
    
        Returns:
            An IPv4Address or IPv6Address object.
    
        Raises:
            ValueError: if the *address* passed isn't either a v4 or a v6
              address
    
        """
        try:
            return IPv4Address(address)
        except (AddressValueError, NetmaskValueError):
            pass
    
        try:
            return IPv6Address(address)
        except (AddressValueError, NetmaskValueError):
            pass
    
>       raise ValueError(f'{address!r} does not appear to be an IPv4 or IPv6 address')
E       ValueError: 'idrac_ip' does not appear to be an IPv4 or IPv6 address

/usr/lib64/python3.13/ipaddress.py:54: ValueError

During handling of the above exception, another exception occurred:

self = <ansible_collections.dellemc.openmanage.tests.unit.plugins.modules.test_idrac_lifecycle_controller_status_info.TestLcStatus object at 0x7fc6a94b2350>
idrac_connection_lcstatus_mock = <MagicMock name='mock.config_mgr' id='140491203745440'>
idrac_default_args = {'_ansible_check_mode': False, '_ansible_keep_remote_files': False, '_ansible_remote_tmp': '/tmp', 'ca_path': '/path/to/ca_cert.pem', ...}
mocker = <pytest_mock.plugin.MockerFixture object at 0x7fc6a84b0ec0>

    def test_main_get_lcstatus_success_case01(self,
                                              idrac_connection_lcstatus_mock,
                                              idrac_default_args,
                                              mocker):
        mocker.patch(MODULE_PATH + "idrac_lifecycle_controller_status_info.IDRACInfo.get_idrac_hw_model",
                     return_value="")
        mocker.patch(MODULE_PATH + "idrac_lifecycle_controller_status_info.IDRACLifecycleControllerStatusInfo.get_lifecycle_controller_status_info",
                     return_value="")
        obj2 = MagicMock()
        idrac_connection_lcstatus_mock.config_mgr = obj2
        type(obj2).LCStatus = PropertyMock(return_value="lcstatus")
        type(obj2).LCReady = PropertyMock(return_value="lcready")
>       result = self._run_module(idrac_default_args)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

tests/unit/plugins/modules/test_idrac_lifecycle_controller_status_info.py:76: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/unit/plugins/modules/common.py:46: in _run_module
    self.module.main()
plugins/modules/idrac_lifecycle_controller_status_info.py:128: in main
    module.fail_json(msg=str(e))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (<ansible.module_utils.basic.AnsibleModule object at 0x7fc6a84b2a50>,)
kwargs = {'failed': True, 'msg': "'idrac_ip' does not appear to be an IPv4 or IPv6 address"}

    def fail_json(*args, **kwargs):
        kwargs['failed'] = True
>       raise AnsibleFailJson(kwargs)
E       ansible_collections.dellemc.openmanage.tests.unit.plugins.modules.utils.AnsibleFailJson: {'msg': "'idrac_ip' does not appear to be an IPv4 or IPv6 address", 'failed': True}

tests/unit/plugins/modules/utils.py:38: AnsibleFailJson
____ TestLcStatus.test_main_get_lcstatus_exception_handling_case[URLError] _____
[gw8] linux -- Python 3.13.5 /home/daniel/Git/Fedora/ansible_collections/dellemc/openmanage/venv/bin/python

    def main():
        specs = {}
        specs.update(idrac_auth_params)
        module = AnsibleModule(
            argument_spec=specs,
            supports_check_mode=True)
        try:
>           with iDRACRedfishAPI(module.params) as idrac:
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

plugins/modules/idrac_lifecycle_controller_status_info.py:111: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
plugins/module_utils/idrac_redfish.py:119: in __init__
    gen_details = self.get_server_generation
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
plugins/module_utils/idrac_redfish.py:224: in get_server_generation
    response = self.invoke_request(MANAGER_URI, 'GET')
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
plugins/module_utils/idrac_redfish.py:187: in invoke_request
    resp = open_url(url, data=data, **url_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/tmp/ansible-test-0stq_dwi/ansible/module_utils/urls.py:995: in open_url
    return Request().open(method, url, data=data, headers=headers, use_proxy=use_proxy,
/tmp/ansible-test-0stq_dwi/ansible/module_utils/urls.py:843: in open
    url, auth_headers, auth_handlers = _configure_auth(url, url_username, url_password, use_gssapi, force_basic_auth, use_netrc)
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/tmp/ansible-test-0stq_dwi/ansible/module_utils/urls.py:640: in _configure_auth
    parsed = urlparse(url)
             ^^^^^^^^^^^^^
/usr/lib64/python3.13/urllib/parse.py:395: in urlparse
    splitresult = urlsplit(url, scheme, allow_fragments)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib64/python3.13/urllib/parse.py:516: in urlsplit
    _check_bracketed_netloc(netloc)
/usr/lib64/python3.13/urllib/parse.py:454: in _check_bracketed_netloc
    _check_bracketed_host(hostname)
/usr/lib64/python3.13/urllib/parse.py:463: in _check_bracketed_host
    ip = ipaddress.ip_address(hostname) # Throws Value Error if not IPv6 or IPv4
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

address = 'idrac_ip'

    def ip_address(address):
        """Take an IP string/int and return an object of the correct type.
    
        Args:
            address: A string or integer, the IP address.  Either IPv4 or
              IPv6 addresses may be supplied; integers less than 2**32 will
              be considered to be IPv4 by default.
    
        Returns:
            An IPv4Address or IPv6Address object.
    
        Raises:
            ValueError: if the *address* passed isn't either a v4 or a v6
              address
    
        """
        try:
            return IPv4Address(address)
        except (AddressValueError, NetmaskValueError):
            pass
    
        try:
            return IPv6Address(address)
        except (AddressValueError, NetmaskValueError):
            pass
    
>       raise ValueError(f'{address!r} does not appear to be an IPv4 or IPv6 address')
E       ValueError: 'idrac_ip' does not appear to be an IPv4 or IPv6 address

/usr/lib64/python3.13/ipaddress.py:54: ValueError

During handling of the above exception, another exception occurred:

self = <ansible_collections.dellemc.openmanage.tests.unit.plugins.modules.test_idrac_lifecycle_controller_status_info.TestLcStatus object at 0x7fc6a93fc140>
exc_type = <class 'urllib.error.URLError'>
idrac_connection_lcstatus_mock = <MagicMock name='mock.config_mgr' id='140491203745440'>
idrac_default_args = {'_ansible_check_mode': False, '_ansible_keep_remote_files': False, '_ansible_remote_tmp': '/tmp', 'ca_path': '/path/to/ca_cert.pem', ...}
mocker = <pytest_mock.plugin.MockerFixture object at 0x7fc6a84b0ec0>

    @pytest.mark.parametrize("exc_type", [RuntimeError, SSLValidationError, ConnectionError, KeyError,
                                          ImportError, ValueError, TypeError, HTTPError, URLError])
    def test_main_get_lcstatus_exception_handling_case(self, exc_type, idrac_connection_lcstatus_mock,
                                                       idrac_default_args, mocker):
        mocker.patch(MODULE_PATH + "idrac_lifecycle_controller_status_info.IDRACInfo.get_idrac_hw_model",
                     return_value="")
        mocker.patch(MODULE_PATH + "idrac_lifecycle_controller_status_info.IDRACLifecycleControllerStatusInfo.get_lifecycle_controller_status_info",
                     return_value="")
        obj2 = MagicMock()
        idrac_connection_lcstatus_mock.config_mgr = obj2
        json_str = to_text(json.dumps({"data": "out"}))
        if exc_type == URLError:
            type(obj2).LCReady = PropertyMock(side_effect=exc_type("url open error"))
>           result = self._run_module(idrac_default_args)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

tests/unit/plugins/modules/test_idrac_lifecycle_controller_status_info.py:93: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/unit/plugins/modules/common.py:46: in _run_module
    self.module.main()
plugins/modules/idrac_lifecycle_controller_status_info.py:128: in main
    module.fail_json(msg=str(e))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (<ansible.module_utils.basic.AnsibleModule object at 0x7fc6a84b2900>,)
kwargs = {'failed': True, 'msg': "'idrac_ip' does not appear to be an IPv4 or IPv6 address"}

    def fail_json(*args, **kwargs):
        kwargs['failed'] = True
>       raise AnsibleFailJson(kwargs)
E       ansible_collections.dellemc.openmanage.tests.unit.plugins.modules.utils.AnsibleFailJson: {'msg': "'idrac_ip' does not appear to be an IPv4 or IPv6 address", 'failed': True}

tests/unit/plugins/modules/utils.py:38: AnsibleFailJson
=============================== warnings summary ===============================
tests/unit/plugins/modules/test_idrac_diagnostics.py: 17 warnings
tests/unit/plugins/modules/test_idrac_lifecycle_controller_job_status_info.py: 4 warnings
tests/unit/plugins/modules/test_idrac_firmware_info.py: 6 warnings
tests/unit/plugins/modules/test_idrac_system_erase.py: 6 warnings
tests/unit/plugins/modules/test_idrac_system_info.py: 6 warnings
tests/unit/plugins/modules/test_idrac_support_assist.py: 26 warnings
tests/unit/plugins/modules/test_ome_application_network_proxy.py: 1 warning
tests/unit/plugins/modules/test_idrac_lifecycle_controller_status_info.py: 11 warnings
  /home/daniel/Git/Fedora/ansible_collections/dellemc/openmanage/plugins/module_utils/utils.py:121: DeprecationWarning: 'count' is passed as positional argument
    ipv6_short = re.sub(r'(:0)+', ':', tempstr, 1)

tests/unit/plugins/modules/test_idrac_server_config_profile.py::TestServerConfigProfile::test_compare_custom_default_configs[params0]
tests/unit/plugins/modules/test_idrac_server_config_profile.py::TestServerConfigProfile::test_compare_custom_default_configs[params0]
  /home/daniel/Git/Fedora/ansible_collections/dellemc/openmanage/plugins/modules/idrac_server_config_profile.py:861: DeprecationWarning: Testing an element's truth value will always return True in future versions.  Use specific 'len(elem)' or 'elem is not None' test instead.
    if component:

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
- generated xml file: /home/daniel/Git/Fedora/ansible_collections/dellemc/openmanage/tests/output/junit/python3.13-modules-units.xml -
=========================== short test summary info ============================
FAILED tests/unit/plugins/modules/test_idrac_lifecycle_controller_job_status_info.py::TestLcJobStatus::test_get_lifecycle_controller_job_status_info_invalid_id - assert {'changed': F...IPv6 address"} == {'changed': F...he job info.'}
FAILED tests/unit/plugins/modules/test_idrac_diagnostics.py::TestDiagnosticsType::test_idrac_diagnostics_main_exception_handling_case[URLError] - KeyError: 'unreachable'
FAILED tests/unit/plugins/modules/test_idrac_support_assist.py::TestSupportAssistType::test_idrac_support_assist_main_exception_handling_case[URLError] - KeyError: 'unreachable'
FAILED tests/unit/plugins/modules/test_idrac_lifecycle_controller_job_status_info.py::TestLcJobStatus::test_get_lifecycle_controller_job_status_info_case02 - assert {'changed': F...IPv6 address"} == {'changed': F...he job info.'}
FAILED tests/unit/plugins/modules/test_idrac_lifecycle_controller_status_info.py::TestLcStatus::test_main_get_lcstatus_success_case00 - ansible_collections.dellemc.openmanage.tests.unit.plugins.modules.utils.Ans...
FAILED tests/unit/plugins/modules/test_idrac_lifecycle_controller_status_info.py::TestLcStatus::test_main_get_lcstatus_success_case01 - ansible_collections.dellemc.openmanage.tests.unit.plugins.modules.utils.Ans...
FAILED tests/unit/plugins/modules/test_idrac_lifecycle_controller_status_info.py::TestLcStatus::test_main_get_lcstatus_exception_handling_case[URLError] - ansible_collections.dellemc.openmanage.tests.unit.plugins.modules.utils.Ans...
```

</details>

### Steps to Reproduce

1) Clone the repo
2) Checkout `9.12.1`
3) Run the tests using `ansible-test`

```
ansible-test --requirements
ansible-test units
```

### Expected Behavior

All the tests should pass.

### Actual Behavior

The tests listed above fail.

### Screenshots

_No response_

### Additional Information

_No response_

--- a/plugins/module_utils/idrac_redfish.py
+++ b/plugins/module_utils/idrac_redfish.py
@@ -238,7 +238,8 @@ class iDRACRedfishAPI(object):
         while job_wait:
             try:
                 response = self.invoke_request(task_uri, "GET")
-                if response.json_data.get("TaskState") == "Running":
+                if response.json_data.get("TaskState") == "Running" or \
+                        response.json_data.get("TaskState") == "New":
                     time.sleep(10)
                 else:
                     break
@@ -306,7 +307,7 @@ class iDRACRedfishAPI(object):
             payload["ShareParameters"]["ProxyUserName"] = share["proxy_username"]
         if share.get("proxy_password") is not None:
             payload["ShareParameters"]["ProxyPassword"] = share["proxy_password"]
-        payload["IncludeInExport"] = include_in_export
+        payload["IncludeInExport"] = [include_in_export]
         response = self.invoke_request(EXPORT_URI, "POST", data=payload)
         if response.status_code == 202 and job_wait:
             task_uri = response.headers["Location"]

--- a/plugins/module_utils/idrac_redfish.py
+++ b/plugins/module_utils/idrac_redfish.py
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 
 # Dell OpenManage Ansible Modules
-# Version 9.12.2
+# Version 9.12.3
 # Copyright (C) 2019-2025 Dell Inc. or its subsidiaries. All Rights Reserved.
 
 # Redistribution and use in source and binary forms, with or without modification,
@@ -51,12 +51,15 @@ idrac_auth_params = {
 
 }
 
+SESSION_RESOURCE_COLLECTION = {
+    "SESSION": "/redfish/v1/SessionService/Sessions",
+    "SESSION_ID": "/redfish/v1/SessionService/Sessions/{Id}",
+}
 MANAGER_URI = "/redfish/v1/Managers/iDRAC.Embedded.1"
 EXPORT_URI = "/redfish/v1/Managers/iDRAC.Embedded.1/Actions/Oem/EID_674_Manager.ExportSystemConfiguration"
 IMPORT_URI = "/redfish/v1/Managers/iDRAC.Embedded.1/Actions/Oem/EID_674_Manager.ImportSystemConfiguration"
 IMPORT_PREVIEW = "/redfish/v1/Managers/iDRAC.Embedded.1/Actions/Oem/EID_674_Manager.ImportSystemConfigurationPreview"
 GET_IDRAC_MANAGER_ATTRIBUTES_9_10 = "/redfish/v1/Managers/iDRAC.Embedded.1/Oem/Dell/DellAttributes/iDRAC.Embedded.1"
-
 
 class OpenURLResponse(object):
     """Handles HTTPResponse"""
@@ -109,24 +112,7 @@ class iDRACRedfishAPI(object):
         self.session_id = None
         self.protocol = 'https'
         self._headers = {'Content-Type': 'application/json', 'Accept': 'application/json'}
-        if self.x_auth_token is not None:
-            self._headers["X-Auth-Token"] = self.x_auth_token
         self.ipaddress = config_ipv6(self.ipaddress)
-        self.SESSION_RESOURCE_COLLECTION = {
-            "SESSION": "/redfish/v1/SessionService/Sessions",
-            "SESSION_ID": "/redfish/v1/SessionService/Sessions/{Id}",
-        }
-        self.SESSION_RESOURCE_COLLECTION = self._get_session_resource_collection(session_dict=self.SESSION_RESOURCE_COLLECTION)
-
-    def _get_session_resource_collection(self, session_dict):
-        gen_details = self.get_server_generation
-        generation = gen_details[0]
-        if generation <= 13:
-            session_dict = {
-                "SESSION": "/redfish/v1/Sessions",
-                "SESSION_ID": "/redfish/v1/Sessions/{Id}",
-            }
-        return session_dict
 
     def _get_url(self, uri):
         return "{0}://{1}:{2}{3}".format(self.protocol, self.ipaddress, self.port, uri)
@@ -167,7 +153,7 @@ class iDRACRedfishAPI(object):
         if headers:
             req_header.update(headers)
         url_kwargs = self._url_common_args_spec(method, api_timeout, headers=headers)
-        if not (path == self.SESSION_RESOURCE_COLLECTION["SESSION"] and method == 'POST'):
+        if not (path == SESSION_RESOURCE_COLLECTION["SESSION"] and method == 'POST'):
             url_kwargs["url_username"] = self.username
             url_kwargs["url_password"] = self.password
             url_kwargs["force_basic_auth"] = True
@@ -199,7 +185,7 @@ class iDRACRedfishAPI(object):
         if self.req_session and not self.x_auth_token:
             payload = {'UserName': self.username,
                        'Password': self.password}
-            path = self.SESSION_RESOURCE_COLLECTION["SESSION"]
+            path = SESSION_RESOURCE_COLLECTION["SESSION"]
             resp = self.invoke_request(path, 'POST', data=payload)
             if resp and resp.success:
                 self.session_id = resp.json_data.get("Id")
@@ -214,7 +200,7 @@ class iDRACRedfishAPI(object):
     def __exit__(self, exc_type, exc_value, traceback):
         """Deletes a session id, which is in use for request"""
         if self.session_id:
-            path = self.SESSION_RESOURCE_COLLECTION["SESSION_ID"].format(Id=self.session_id)
+            path = SESSION_RESOURCE_COLLECTION["SESSION_ID"].format(Id=self.session_id)
             self.invoke_request(path, 'DELETE')
         return False
 
@@ -240,6 +226,7 @@ class iDRACRedfishAPI(object):
 
         return generation, firmware_version, hw_model
 
+
     def wait_for_job_complete(self, task_uri, job_wait=False):
         """
         This function wait till the job completion.
@@ -251,7 +238,7 @@ class iDRACRedfishAPI(object):
         while job_wait:
             try:
                 response = self.invoke_request(task_uri, "GET")
-                if response.json_data.get("TaskState") == "Running" or response.json_data.get("TaskState") == "New":
+                if response.json_data.get("TaskState") == "Running":
                     time.sleep(10)
                 else:
                     break
@@ -319,7 +306,7 @@ class iDRACRedfishAPI(object):
             payload["ShareParameters"]["ProxyUserName"] = share["proxy_username"]
         if share.get("proxy_password") is not None:
             payload["ShareParameters"]["ProxyPassword"] = share["proxy_password"]
-        payload["IncludeInExport"] = [include_in_export]
+        payload["IncludeInExport"] = include_in_export
         response = self.invoke_request(EXPORT_URI, "POST", data=payload)
         if response.status_code == 202 and job_wait:
             task_uri = response.headers["Location"]

--- a/plugins/module_utils/idrac_redfish.py
+++ b/plugins/module_utils/idrac_redfish.py
@@ -61,6 +61,7 @@ IMPORT_URI = "/redfish/v1/Managers/iDRAC.Embedded.1/Actions/Oem/EID_674_Manager.
 IMPORT_PREVIEW = "/redfish/v1/Managers/iDRAC.Embedded.1/Actions/Oem/EID_674_Manager.ImportSystemConfigurationPreview"
 GET_IDRAC_MANAGER_ATTRIBUTES_9_10 = "/redfish/v1/Managers/iDRAC.Embedded.1/Oem/Dell/DellAttributes/iDRAC.Embedded.1"
 
+
 class OpenURLResponse(object):
     """Handles HTTPResponse"""
 
@@ -225,7 +226,6 @@ class iDRACRedfishAPI(object):
             hw_model = "iDRAC 8"
 
         return generation, firmware_version, hw_model
-
 
     def wait_for_job_complete(self, task_uri, job_wait=False):
         """

--- a/plugins/module_utils/redfish.py
+++ b/plugins/module_utils/redfish.py
@@ -57,6 +57,7 @@ HOST_UNRESOLVED_MSG = "Unable to resolve hostname or IP {0}."
 MANAGER_URI = "/redfish/v1/Managers/iDRAC.Embedded.1"
 GET_IDRAC_MANAGER_ATTRIBUTES_9_10 = "/redfish/v1/Managers/iDRAC.Embedded.1/Oem/Dell/DellAttributes/iDRAC.Embedded.1"
 
+
 class OpenURLResponse(object):
     """Handles HTTPResponse"""
 
@@ -236,7 +237,6 @@ class Redfish(object):
         except HTTPError:
             hw_model = "iDRAC 8"
         return generation, firmware_version, hw_model
-
 
     def strip_substr_dict(self, odata_dict, chkstr='@odata.'):
         cp = odata_dict.copy()

--- a/plugins/module_utils/redfish.py
+++ b/plugins/module_utils/redfish.py
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 
 # Dell OpenManage Ansible Modules
-# Version 9.12.2
+# Version 9.12.3
 # Copyright (C) 2019-2025 Dell Inc. or its subsidiaries. All Rights Reserved.
 
 # Redistribution and use in source and binary forms, with or without modification,
@@ -48,10 +48,14 @@ redfish_auth_params = {
     "timeout": {"type": "int", "default": 30},
 }
 
+SESSION_RESOURCE_COLLECTION = {
+    "SESSION": "/redfish/v1/SessionService/Sessions",
+    "SESSION_ID": "/redfish/v1/SessionService/Sessions/{Id}",
+}
+
 HOST_UNRESOLVED_MSG = "Unable to resolve hostname or IP {0}."
 MANAGER_URI = "/redfish/v1/Managers/iDRAC.Embedded.1"
 GET_IDRAC_MANAGER_ATTRIBUTES_9_10 = "/redfish/v1/Managers/iDRAC.Embedded.1/Oem/Dell/DellAttributes/iDRAC.Embedded.1"
-
 
 class OpenURLResponse(object):
     """Handles HTTPResponse"""
@@ -70,6 +74,10 @@ class OpenURLResponse(object):
             raise ValueError("Unable to parse json")
 
     @property
+    def status_code(self):
+        return self.resp.getcode()
+
+    @property
     def success(self):
         status = self.status_code
         return status >= 200 & status <= 299
@@ -77,10 +85,6 @@ class OpenURLResponse(object):
     @property
     def headers(self):
         return self.resp.headers
-
-    @property
-    def status_code(self):
-        return self.resp.getcode()
 
     @property
     def reason(self):
@@ -106,23 +110,20 @@ class Redfish(object):
         self.root_uri = '/redfish/v1/'
         self._headers = {'Content-Type': 'application/json', 'Accept': 'application/json'}
         self.hostname = config_ipv6(self.hostname)
-        if self.x_auth_token is not None:
-            self._headers["X-Auth-Token"] = self.x_auth_token
-        self.SESSION_RESOURCE_COLLECTION = {
-            "SESSION": "/redfish/v1/SessionService/Sessions",
-            "SESSION_ID": "/redfish/v1/SessionService/Sessions/{Id}",
-        }
-        gen_details = self.get_server_generation
-        generation = gen_details[0]
-        if generation <= 13:
-            self.SESSION_RESOURCE_COLLECTION = {
-                "SESSION": "/redfish/v1/Sessions",
-                "SESSION_ID": "/redfish/v1/Sessions/{Id}",
-            }
 
     def _get_base_url(self):
         """builds base url"""
         return '{0}://{1}'.format(self.protocol, self.hostname)
+
+    def _build_url(self, path, query_param=None):
+        """builds complete url"""
+        url = path
+        base_uri = self._get_base_url()
+        if path:
+            url = base_uri + path
+        if query_param:
+            url += "?{0}".format(urlencode(query_param))
+        return url
 
     def _url_common_args_spec(self, method, api_timeout, headers=None):
         """Creates an argument common spec"""
@@ -144,23 +145,13 @@ class Redfish(object):
         }
         return url_kwargs
 
-    def _build_url(self, path, query_param=None):
-        """builds complete url"""
-        url = path
-        base_uri = self._get_base_url()
-        if path:
-            url = base_uri + path
-        if query_param:
-            url += "?{0}".format(urlencode(query_param))
-        return url
-
     def _args_without_session(self, path, method, api_timeout, headers=None):
         """Creates an argument spec in case of basic authentication"""
         req_header = self._headers
         if headers:
             req_header.update(headers)
         url_kwargs = self._url_common_args_spec(method, api_timeout, headers=headers)
-        if not (path == self.SESSION_RESOURCE_COLLECTION["SESSION"] and method == 'POST'):
+        if not (path == SESSION_RESOURCE_COLLECTION["SESSION"] and method == 'POST'):
             url_kwargs["url_username"] = self.username
             url_kwargs["url_password"] = self.password
             url_kwargs["force_basic_auth"] = True
@@ -207,7 +198,7 @@ class Redfish(object):
         if self.req_session and not self.x_auth_token:
             payload = {'UserName': self.username,
                        'Password': self.password}
-            path = self.SESSION_RESOURCE_COLLECTION["SESSION"]
+            path = SESSION_RESOURCE_COLLECTION["SESSION"]
             resp = self.invoke_request('POST', path, data=payload)
             if resp and resp.success:
                 self.session_id = resp.json_data.get("Id")
@@ -222,7 +213,7 @@ class Redfish(object):
     def __exit__(self, exc_type, exc_value, traceback):
         """Deletes a session id, which is in use for request for redfish session"""
         if self.session_id:
-            path = self.SESSION_RESOURCE_COLLECTION["SESSION_ID"].format(Id=self.session_id)
+            path = SESSION_RESOURCE_COLLECTION["SESSION_ID"].format(Id=self.session_id)
             self.invoke_request('DELETE', path)
         return False
 
@@ -245,6 +236,7 @@ class Redfish(object):
         except HTTPError:
             hw_model = "iDRAC 8"
         return generation, firmware_version, hw_model
+
 
     def strip_substr_dict(self, odata_dict, chkstr='@odata.'):
         cp = odata_dict.copy()
@@ -282,12 +274,12 @@ class RedfishAnsibleModule(AnsibleModule):
         if mutually_exclusive is None:
             mutually_exclusive = []
         mutually_exclusive.extend(auth_mutually_exclusive)
-        if required_one_of is None:
-            required_one_of = []
-        required_one_of.extend(auth_required_one_of)
         if required_together is None:
             required_together = []
         required_together.extend(auth_required_together)
+        if required_one_of is None:
+            required_one_of = []
+        required_one_of.extend(auth_required_one_of)
         if required_by is None:
             required_by = {}
 

--- a/plugins/module_utils/redfish.py
+++ b/plugins/module_utils/redfish.py
@@ -75,10 +75,6 @@ class OpenURLResponse(object):
             raise ValueError("Unable to parse json")
 
     @property
-    def status_code(self):
-        return self.resp.getcode()
-
-    @property
     def success(self):
         status = self.status_code
         return status >= 200 & status <= 299
@@ -86,6 +82,10 @@ class OpenURLResponse(object):
     @property
     def headers(self):
         return self.resp.headers
+
+    @property
+    def status_code(self):
+        return self.resp.getcode()
 
     @property
     def reason(self):
@@ -116,16 +116,6 @@ class Redfish(object):
         """builds base url"""
         return '{0}://{1}'.format(self.protocol, self.hostname)
 
-    def _build_url(self, path, query_param=None):
-        """builds complete url"""
-        url = path
-        base_uri = self._get_base_url()
-        if path:
-            url = base_uri + path
-        if query_param:
-            url += "?{0}".format(urlencode(query_param))
-        return url
-
     def _url_common_args_spec(self, method, api_timeout, headers=None):
         """Creates an argument common spec"""
         req_header = self._headers
@@ -145,6 +135,16 @@ class Redfish(object):
             "follow_redirects": 'all',
         }
         return url_kwargs
+
+    def _build_url(self, path, query_param=None):
+        """builds complete url"""
+        url = path
+        base_uri = self._get_base_url()
+        if path:
+            url = base_uri + path
+        if query_param:
+            url += "?{0}".format(urlencode(query_param))
+        return url
 
     def _args_without_session(self, path, method, api_timeout, headers=None):
         """Creates an argument spec in case of basic authentication"""
@@ -274,12 +274,12 @@ class RedfishAnsibleModule(AnsibleModule):
         if mutually_exclusive is None:
             mutually_exclusive = []
         mutually_exclusive.extend(auth_mutually_exclusive)
-        if required_together is None:
-            required_together = []
-        required_together.extend(auth_required_together)
         if required_one_of is None:
             required_one_of = []
         required_one_of.extend(auth_required_one_of)
+        if required_together is None:
+            required_together = []
+        required_together.extend(auth_required_together)
         if required_by is None:
             required_by = {}
 

--- a/tests/unit/plugins/module_utils/test_idrac_redfish.py
+++ b/tests/unit/plugins/module_utils/test_idrac_redfish.py
@@ -31,7 +31,6 @@ API_TASK = '/api/tasks'
 SLEEP_TIME = 'idrac_redfish.time.sleep'
 MANAGER_URI = "/redfish/v1/Managers/iDRAC.Embedded.1"
 GET_IDRAC_MANAGER_ATTRIBUTES_9_10 = "/redfish/v1/Managers/iDRAC.Embedded.1/Oem/Dell/DellAttributes/iDRAC.Embedded.1"
-#SESSION_FUNCTION = 'idrac_redfish.iDRACRedfishAPI._get_session_resource_collection'
 RESP = {
     "Model": "17G",
     "FirmwareVersion": "1.20.30"

--- a/tests/unit/plugins/module_utils/test_idrac_redfish.py
+++ b/tests/unit/plugins/module_utils/test_idrac_redfish.py
@@ -2,7 +2,7 @@
 
 #
 # Dell OpenManage Ansible Modules
-# Version 9.12.2
+# Version 9.12.3
 # Copyright (C) 2023-2025 Dell Inc.
 
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
@@ -17,7 +17,7 @@ __metaclass__ = type
 import pytest
 from ansible.module_utils.urls import ConnectionError, SSLValidationError
 from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
-from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, OpenURLResponse, IdracAnsibleModule
+from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, OpenURLResponse
 from unittest.mock import MagicMock
 import json
 import os
@@ -31,7 +31,7 @@ API_TASK = '/api/tasks'
 SLEEP_TIME = 'idrac_redfish.time.sleep'
 MANAGER_URI = "/redfish/v1/Managers/iDRAC.Embedded.1"
 GET_IDRAC_MANAGER_ATTRIBUTES_9_10 = "/redfish/v1/Managers/iDRAC.Embedded.1/Oem/Dell/DellAttributes/iDRAC.Embedded.1"
-SESSION_FUNCTION = 'idrac_redfish.iDRACRedfishAPI._get_session_resource_collection'
+#SESSION_FUNCTION = 'idrac_redfish.iDRACRedfishAPI._get_session_resource_collection'
 RESP = {
     "Model": "17G",
     "FirmwareVersion": "1.20.30"
@@ -68,27 +68,11 @@ class TestIdracRedfishRest(object):
         return module_parameters
 
     @pytest.fixture
-    def idrac_redfish_object(self, module_params, mocker):
-        resp = {
-            "SESSION": SESSION_10,
-            "SESSION_ID": SESSION_ID_10,
-        }
-        mocker.patch(MODULE_UTIL_PATH + SESSION_FUNCTION,
-                     return_value=resp)
+    def idrac_redfish_object(self, module_params):
         idrac_redfish_obj = iDRACRedfishAPI(module_params)
         return idrac_redfish_obj
 
-    @pytest.fixture
-    def arg_spec(self):
-        argument_specifications = {'resource_id': 'sample_resource'}
-        return argument_specifications
-
-    @pytest.fixture
-    def idrac_ansible_object(self, arg_spec):
-        idrac_ansible_obj = IdracAnsibleModule(arg_spec)
-        return idrac_ansible_obj
-
-    def test_invoke_request_with_session(self, mock_response, mocker, module_params, idrac_redfish_object):
+    def test_invoke_request_with_session(self, mock_response, mocker, module_params):
         mocker.patch(MODULE_UTIL_PATH + OPEN_URL,
                      return_value=mock_response)
         req_session = True
@@ -104,12 +88,6 @@ class TestIdracRedfishRest(object):
         module_params = {'idrac_ip': '2001:db8:3333:4444:5555:6666:7777:8888', 'idrac_user': 'username',
                          'idrac_password': 'password', "idrac_port": '443'}
         req_session = False
-        resp = {
-            "SESSION": SESSION_10,
-            "SESSION_ID": SESSION_ID_10,
-        }
-        mocker.patch(MODULE_UTIL_PATH + SESSION_FUNCTION,
-                     return_value=resp)
         with iDRACRedfishAPI(module_params, req_session) as obj:
             response = obj.invoke_request(TEST_PATH, "GET")
         assert response.status_code == 200
@@ -119,12 +97,6 @@ class TestIdracRedfishRest(object):
     def test_invoke_request_without_session_with_header(self, mock_response, mocker, module_params):
         mocker.patch(MODULE_UTIL_PATH + OPEN_URL,
                      return_value=mock_response)
-        resp = {
-            "SESSION": SESSION_10,
-            "SESSION_ID": SESSION_ID_10,
-        }
-        mocker.patch(MODULE_UTIL_PATH + SESSION_FUNCTION,
-                     return_value=resp)
         req_session = False
         with iDRACRedfishAPI(module_params, req_session) as obj:
             response = obj.invoke_request(TEST_PATH, "POST", headers={
@@ -162,6 +134,34 @@ class TestIdracRedfishRest(object):
         with pytest.raises(HTTPError):
             with iDRACRedfishAPI(module_params, req_session) as obj:
                 obj.invoke_request(TEST_PATH, "GET")
+
+    def test_get_idrac_local_account_attr(self, idrac_redfish_object):
+        idrac_attrs = {
+            "SystemConfiguration": {
+                "Components": [
+                    {
+                        "FQDD": "iDRAC.Embedded.1",
+                        "Attributes": [
+                            {
+                                "Name": "Users.1",
+                                "Value": 1
+                            },
+                            {
+                                "Name": "Users.2",
+                                "Value": 2
+                            },
+                            {
+                                "Name": "System1",
+                                "Value": "system_data"
+                            }
+                        ]
+                    }
+                ]
+            }
+        }
+        result = idrac_redfish_object.get_idrac_local_account_attr(
+            idrac_attribues=idrac_attrs, fqdd="iDRAC.Embedded.1")
+        assert result == {'Users.1': 1, 'Users.2': 2}
 
     @pytest.mark.parametrize("query_params", [
         {"inp": {"$filter": "UserName eq 'admin'"},
@@ -235,34 +235,6 @@ class TestIdracRedfishRest(object):
         with pytest.raises(ValueError):
             with iDRACRedfishAPI(module_params, True) as obj:
                 obj.wait_for_job_complete(API_TASK, True)
-
-    def test_get_idrac_local_account_attr(self, idrac_redfish_object):
-        idrac_attrs = {
-            "SystemConfiguration": {
-                "Components": [
-                    {
-                        "FQDD": "iDRAC.Embedded.1",
-                        "Attributes": [
-                            {
-                                "Name": "Users.1",
-                                "Value": 1
-                            },
-                            {
-                                "Name": "Users.2",
-                                "Value": 2
-                            },
-                            {
-                                "Name": "System1",
-                                "Value": "system_data"
-                            }
-                        ]
-                    }
-                ]
-            }
-        }
-        result = idrac_redfish_object.get_idrac_local_account_attr(
-            idrac_attribues=idrac_attrs, fqdd="iDRAC.Embedded.1")
-        assert result == {'Users.1': 1, 'Users.2': 2}
 
     @pytest.mark.parametrize("inp_data", [
         {

--- a/tests/unit/plugins/module_utils/test_redfish.py
+++ b/tests/unit/plugins/module_utils/test_redfish.py
@@ -20,7 +20,7 @@ import sys
 from ansible.module_utils.urls import ConnectionError, SSLValidationError
 from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
 from ansible_collections.dellemc.openmanage.plugins.module_utils.redfish \
-  import Redfish, OpenURLResponse, RedfishAnsibleModule
+    import Redfish, OpenURLResponse, RedfishAnsibleModule
 from unittest.mock import MagicMock, patch
 import json
 
@@ -28,6 +28,7 @@ MODULE_UTIL_PATH = 'ansible_collections.dellemc.openmanage.plugins.module_utils.
 OPEN_URL = 'redfish.open_url'
 TEST_PATH = "/testpath"
 MANAGER_URI = "/redfish/v1/Managers/iDRAC.Embedded.1"
+
 
 class TestRedfishRest(object):
 
@@ -57,11 +58,10 @@ class TestRedfishRest(object):
         obj.success = True
         if 'path' in kwargs and kwargs['path'] == MANAGER_URI:
             obj.json_data = {'Model': '17G Monolithic',
-                            'FirmwareVersion': 'x.x.x.x'}
+                             'FirmwareVersion': 'x.x.x.x'}
         else:
             obj.json_data = {'Attributes': {'Info.1.HWModel': 'iDRAC 10'}}
         return obj
-
 
     def test_get_server_generation(self, mocker, module_params):
         mocker.patch(MODULE_UTIL_PATH + 'redfish.Redfish.invoke_request', self.mock_get_dynamic_redfish_invoke_request)
@@ -71,28 +71,26 @@ class TestRedfishRest(object):
             response = obj.get_server_generation
         assert response == (17, 'x.x.x.x', 'iDRAC 10')
 
-
     def test_invoke_request_with_session_with_header(self, mock_response, mocker, module_params):
         Redfish.get_server_generation = [12]
         mocker.patch(MODULE_UTIL_PATH + OPEN_URL,
-                    return_value=mock_response)
+                     return_value=mock_response)
         req_session = False
         module_params.update({'x_auth_token': 'token_id'})
         with Redfish(module_params, req_session) as obj:
             response = obj.invoke_request(TEST_PATH, "GET", headers={
-                                        "application": "octstream"})
+                                          "application": "octstream"})
         assert response.status_code == 200
         assert response.json_data == {"value": "data"}
         assert response.success is True
 
-
     def test_invoke_request_without_session(self, mock_response, mocker):
         mocker.patch(MODULE_UTIL_PATH + OPEN_URL,
-                    return_value=mock_response)
+                     return_value=mock_response)
         mocker.patch(MODULE_UTIL_PATH + OPEN_URL,
-                    return_value=mock_response)
+                     return_value=mock_response)
         module_params = {'baseuri': '[2001:db8:3333:4444:5555:6666:7777:8888]:443', 'username': 'username',
-                        'password': 'password', "port": 443}
+                         'password': 'password', "port": 443}
         req_session = False
         with Redfish(module_params, req_session) as obj:
             response = obj.invoke_request(TEST_PATH, "GET")

--- a/tests/unit/plugins/module_utils/test_redfish.py
+++ b/tests/unit/plugins/module_utils/test_redfish.py
@@ -2,7 +2,7 @@
 
 #
 # Dell OpenManage Ansible Modules
-# Version 9.12.2
+# Version 9.12.3
 # Copyright (C) 2023-2025 Dell Inc.
 
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
@@ -15,17 +15,19 @@ from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
 import pytest
+import io
+import sys
 from ansible.module_utils.urls import ConnectionError, SSLValidationError
 from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
-from ansible_collections.dellemc.openmanage.plugins.module_utils.redfish import Redfish, OpenURLResponse, RedfishAnsibleModule
-from unittest.mock import MagicMock
+from ansible_collections.dellemc.openmanage.plugins.module_utils.redfish \
+  import Redfish, OpenURLResponse, RedfishAnsibleModule
+from unittest.mock import MagicMock, patch
 import json
 
 MODULE_UTIL_PATH = 'ansible_collections.dellemc.openmanage.plugins.module_utils.'
 OPEN_URL = 'redfish.open_url'
 TEST_PATH = "/testpath"
 MANAGER_URI = "/redfish/v1/Managers/iDRAC.Embedded.1"
-
 
 class TestRedfishRest(object):
 
@@ -55,10 +57,11 @@ class TestRedfishRest(object):
         obj.success = True
         if 'path' in kwargs and kwargs['path'] == MANAGER_URI:
             obj.json_data = {'Model': '17G Monolithic',
-                             'FirmwareVersion': 'x.x.x.x'}
+                            'FirmwareVersion': 'x.x.x.x'}
         else:
             obj.json_data = {'Attributes': {'Info.1.HWModel': 'iDRAC 10'}}
         return obj
+
 
     def test_get_server_generation(self, mocker, module_params):
         mocker.patch(MODULE_UTIL_PATH + 'redfish.Redfish.invoke_request', self.mock_get_dynamic_redfish_invoke_request)
@@ -68,22 +71,47 @@ class TestRedfishRest(object):
             response = obj.get_server_generation
         assert response == (17, 'x.x.x.x', 'iDRAC 10')
 
+
     def test_invoke_request_with_session_with_header(self, mock_response, mocker, module_params):
         Redfish.get_server_generation = [12]
         mocker.patch(MODULE_UTIL_PATH + OPEN_URL,
-                     return_value=mock_response)
+                    return_value=mock_response)
         req_session = False
         module_params.update({'x_auth_token': 'token_id'})
         with Redfish(module_params, req_session) as obj:
             response = obj.invoke_request(TEST_PATH, "GET", headers={
-                                          "application": "octstream"})
+                                        "application": "octstream"})
+        assert response.status_code == 200
+        assert response.json_data == {"value": "data"}
+        assert response.success is True
+
+
+    def test_invoke_request_without_session(self, mock_response, mocker):
+        mocker.patch(MODULE_UTIL_PATH + OPEN_URL,
+                    return_value=mock_response)
+        mocker.patch(MODULE_UTIL_PATH + OPEN_URL,
+                    return_value=mock_response)
+        module_params = {'baseuri': '[2001:db8:3333:4444:5555:6666:7777:8888]:443', 'username': 'username',
+                        'password': 'password', "port": 443}
+        req_session = False
+        with Redfish(module_params, req_session) as obj:
+            response = obj.invoke_request(TEST_PATH, "GET")
+        assert response.status_code == 200
+        assert response.json_data == {"value": "data"}
+        assert response.success is True
+        assert response.headers == {"X-Auth-Token": "token_id"}
+
+    def test_invoke_request_with_session(self, mock_response, mocker, module_params):
+        mocker.patch(MODULE_UTIL_PATH + OPEN_URL,
+                     return_value=mock_response)
+        req_session = True
+        with Redfish(module_params, req_session) as obj:
+            response = obj.invoke_request(TEST_PATH, "GET")
         assert response.status_code == 200
         assert response.json_data == {"value": "data"}
         assert response.success is True
 
     def test_invoke_request_without_session(self, mock_response, mocker):
-        mocker.patch(MODULE_UTIL_PATH + OPEN_URL,
-                     return_value=mock_response)
         mocker.patch(MODULE_UTIL_PATH + OPEN_URL,
                      return_value=mock_response)
         module_params = {'baseuri': '[2001:db8:3333:4444:5555:6666:7777:8888]:443', 'username': 'username',
@@ -94,7 +122,6 @@ class TestRedfishRest(object):
         assert response.status_code == 200
         assert response.json_data == {"value": "data"}
         assert response.success is True
-        assert response.headers == {"X-Auth-Token": "token_id"}
 
     def test_invoke_request_without_session_with_header(self, mock_response, mocker, module_params):
         mocker.patch(MODULE_UTIL_PATH + OPEN_URL,
@@ -197,10 +224,21 @@ class TestRedfishRest(object):
         assert reason_ret == "returning reason"
 
 
-class TestRedfishAnsibleModule(object):
-    def test_call_class_redfish_ansible_module(self):
-        random_value = {'value': 'data'}
-        with pytest.raises(SystemExit) as ex:
-            RedfishAnsibleModule(random_value, bypass_checks=True)
-        # Asserting only this as class has only __init__ method
-        assert ex.type == SystemExit
+class TestRedfishAnsibleModule:
+    def test_redfish_module_initialization(self):
+        # Simulate Ansible JSON input
+        fake_input = json.dumps({
+            "baseuri": "https://example.com",
+            "username": "admin",
+            "password": "secret"
+        }).encode('utf-8')
+
+        # Create a binary stream and wrap it with a text wrapper
+        fake_stdin = io.TextIOWrapper(io.BytesIO(fake_input), encoding='utf-8')
+
+        with patch.object(sys, 'stdin', fake_stdin):
+            with pytest.raises(SystemExit) as ex:
+                RedfishAnsibleModule(argument_spec={}, bypass_checks=True)
+
+            # AnsibleModule exits after parsing input
+            assert ex.type == SystemExit

